File: isegoria/src/main/java/com/eulersbridge/isegoria/models/NewsArticle.java
package com.eulersbridge.isegoria.models;

import com.google.gson.annotations.SerializedName;

import java.util.List;

/**
 * Created by Seb on 02/11/2017.
 */

@org.parceler.Parcel
public class NewsArticle {

    @SerializedName("articleId")
    public long id;

    public long institutionId;
    public String title;
    public String content;

    public List<Photo> photos;

    public long dateTimestamp;

    @SerializedName("likes")
    public int likeCount = 0;

    @SerializedName("creatorProfile")
    public UserProfile creator;

    @SerializedName("inappropriateContent")
    public boolean hasInappropriateContent;
}

File: isegoria/src/main/kotlin/com/eulersbridge/isegoria/network/api/model/NewsArticle.kt
package com.eulersbridge.isegoria.network.api.model

import android.annotation.SuppressLint
import android.os.Parcelable
import com.eulersbridge.isegoria.network.adapters.Timestamp
import com.squareup.moshi.Json
import kotlinx.android.parcel.Parcelize

@SuppressLint("ParcelCreator")
@Parcelize
data class NewsArticle(
        @field:Json(name = "articleId")
        val id: Long,

        val institutionId: Long?,
        val title: String,
        val content: String,
        val photos: List<Photo>?,

        @Timestamp
        val date: Long,

        @field:Json(name = "likes")
        val likeCount: Int = 0,

        val creatorEmail: String?,

        @field:Json(name = "creatorProfile")
        val creator: Contact,

        @field:Json(name = "inappropriateContent")
        val hasInappropriateContent: Boolean

) : Parcelable {
    internal fun getPhotoUrl()
        = photos?.firstOrNull()?.getPhotoUrl()
}